src/main/java/com/troubleshooting/troubleshootingtool/ContainerDataRetrieval.java

```diff
@@ -0,0 +1,69 @@
+package com.troubleshooting.troubleshootingtool;
+
+import com.troubleshooting.model.ConjurLogModel;
+import com.troubleshooting.model.EnvModel;
+
+import com.troubleshooting.model.EnvsModel;
+import com.troubleshooting.model.LogsModel;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+// Retrieves the data returned from the Docker Client and packs the data in the appropriate data models
+@Component
+public class ContainerDataRetrieval implements IDataRetrieval {
+    ContainerDataAccess access = new ContainerDataAccess();
+
+    ConjurLogModel log = new ConjurLogModel();
+    LogsModel logs = new LogsModel();
+
+    EnvModel env = new EnvModel();
+    EnvsModel envs = new EnvsModel();
+
+    //	Extract Logs
+    @Override
+    public LogsModel getData(String containerID) {
+        List<String> loggingFrames = access.getLogs(containerID);
```

Comment:
You are certain that access.getLogs wii not return null and won't trow an exception?

Reply:
If a container exists then there should be at startup logs. In terms of what to return, would I rethrow the null pointer or return an empty object? 
```
try {
}  catch (NullPointerException e) {
  throw e
}
```
or an empty object?
```
try {
}  catch (NullPointerException e) {
      return new LogsModel();
}
```

Reply:
Better to check for null rather than try and fall.

Reply:
check for null or throw null?

Reply:
It depends on the logic. In java there are two types of exceptions:
1. RuntimeException (means that something unexpected had happened that the user cannot be recovered)
2. CheckedException (you must try or catch or declare that the method throws)
Usually one don't throw a RuntimeExcpetion and also not Java API exception but creates a BusinessExecption
```
class UserCreateException extends Exception {}
try {
  registerUser();
} catch (Throwabe t) {. // Throwable is super class of runtime and checked excepitons
  throw new UserCreateException("Failed to create user", t);
}
```